tests/subsys/bluetooth/mesh/light_hue/src/main.c

```diff
@@ -0,0 +1,688 @@
+		ztest_expect_value(hue_set, set->transition->time, set->transition->time);
+		ztest_expect_value(hue_set, set->direction, set->direction);
+	}
+}
+
+static void expect_hue_move_set_cb(struct bt_mesh_light_hue_move *move)
+{
+	ztest_expect_value(hue_move_set, move->delta, move->delta);
+
+	if (move->transition) {
+		ztest_expect_value(hue_move_set, move->transition->time, move->transition->time);
+	}
+}
+
+static void setup(void *f)
+{
+	zassert_not_null(_bt_mesh_light_hue_srv_cb.init, "Init cb is null");
+	zassert_ok(_bt_mesh_light_hue_srv_cb.init(&mock_light_hue_srv_model), "Init failed");
+}
+
+static void teardown(void *f)
+{
+	zassert_not_null(_bt_mesh_light_hue_srv_cb.reset, "Reset cb is null");
+	_bt_mesh_light_hue_srv_cb.reset(&mock_light_hue_srv_model);
+}
+
+/* Test Hue value clamping without transition. */
+ZTEST(light_hue_srv_test, test_hue_set)
+{
+	struct {
+		uint16_t min;
+		uint16_t max;
+		uint16_t current_hue;
+		uint16_t target_hue;
+		uint16_t expected_hue;
+	} test_vector[] = {
+		/* Range Min, Range Max, Current, Target, Expected */
+		{  0,         0xFFFF,    0,       0,      0       },
+		{  0,         0xFFFF,    0,       0xFFFF, 0xFFFF  },
+		/* [100, 0xFF00] */
+		{  100,       0xFF00,    0xFFFF,  99,     100     },
+		{  100,       0xFF00,    100,     0xFF01, 0xFF00  },
+		{  100,       0xFF00,    0xFF00,  101,    101     },
+		{  100,       0xFF00,    101,     0xFEFF, 0xFEFF  },
+		/* [0 - 100, 0xFF00 - 0xFFFF] */
+		{  0xFF00,    100,       0xFEFF,  101,    100     },
+		{  0xFF00,    100,       100,     0xFEFF, 0xFF00  },
+		{  0xFF00,    100,       0xFF00,  99,     99      },
+		{  0xFF00,    100,       99,      0xFF01, 0xFF01  },
+	};
+
+	for (size_t i = 0; i < ARRAY_SIZE(test_vector); i++) {
+		struct bt_mesh_light_hue_status status = {};
+
+		printk("Checking test_vector[%d]\n", i);
+
+		light_hue_srv.range.min = test_vector[i].min;
+		light_hue_srv.range.max = test_vector[i].max;
+		light_hue_val = test_vector[i].current_hue;
+
+		status.current = test_vector[i].expected_hue;
+		expect_hue_status_pub(&status);
+
+		struct bt_mesh_light_hue hue_set = {
+			.lvl = test_vector[i].target_hue,
+		};
+		struct bt_mesh_light_hue exp_hue_set = {
+			.lvl = test_vector[i].expected_hue,
+		};
+
+		expect_hue_set_cb(&exp_hue_set);
+		zassert_ok(hue_set_unack_send(&hue_set));
+	}
+}
+
+/* Test Hue value clamping with transition. */
+ZTEST(light_hue_srv_test, test_hue_set_transition)
+{
+	struct {
+		uint16_t min;
+		uint16_t max;
+		uint16_t current_hue;
+		uint16_t target_hue;
+		uint16_t expected_hue;
+		int direction;
+	} test_vector[] = {
+		/* Range Min, Range Max, Current, Target, Expected, Diretion */
+		{  0,         0xFFFF,    0,       0,      0,         1  },
+		{  0,         0xFFFF,    0,       0xFFFF, 0xFFFF,   -1  },
+		/* Direction points to the shortest path. */
+		{  0,         0xFFFF,    0xFFFF,  0,      0,         1  },
+		{  0,         0xFFFF,    0,       0xFFFF, 0xFFFF,   -1  },
```

Comment:
There is no point in testing this vector twice, is it? It's covered on line 331.